core/trino-main/src/test/java/io/trino/type/TestLongTimestampWithTimeZoneType.java

```diff
@@ -114,58 +114,78 @@ public void testNextValue()
+        LongTimestampWithTimeZone sequenceValueType = LongTimestampWithTimeZone.fromEpochMillisAndFraction(middleRangeEpochMillis, 0, UTC_KEY);
+        LongTimestampWithTimeZone minValueType = LongTimestampWithTimeZone.fromEpochMillisAndFraction(minValue, 0, UTC_KEY);
+        LongTimestampWithTimeZone nextToMinValueType = LongTimestampWithTimeZone.fromEpochMillisAndFraction(minValue, step, UTC_KEY);
+        LongTimestampWithTimeZone previousToMaxValueType = LongTimestampWithTimeZone.fromEpochMillisAndFraction(maxValue, oneMsInPicos - (2 * step), UTC_KEY);
+        LongTimestampWithTimeZone maxValueType = LongTimestampWithTimeZone.fromEpochMillisAndFraction(maxValue, oneMsInPicos - step, UTC_KEY);
+
+        assertThat(type.getNextValue(minValueType))
+                .isEqualTo(Optional.of(nextToMinValueType));
+        assertThat(type.getNextValue(nextToMinValueType))
+                .isEqualTo(Optional.of(LongTimestampWithTimeZone.fromEpochMillisAndFraction(minValue, 2 * step, UTC_KEY)));
+
+        assertThat(type.getNextValue(zeroValueType))
+                .isEqualTo(Optional.of(LongTimestampWithTimeZone.fromEpochMillisAndFraction(0, step, UTC_KEY)));
+        assertThat(type.getNextValue(sequenceValueType))
+                .isEqualTo(Optional.of(LongTimestampWithTimeZone.fromEpochMillisAndFraction(middleRangeEpochMillis, step, UTC_KEY)));
+
+        assertThat(type.getNextValue(previousToMaxValueType))
+                .isEqualTo(Optional.of(maxValueType));
+        assertThat(type.getNextValue(maxValueType))
                 .isEqualTo(Optional.empty());
     }
 
     public static Stream<Arguments> testPreviousNextValueEveryPrecisionDataProvider()
     {
         return Stream.of(
-                Arguments.of(0, Long.MIN_VALUE + 775808, Long.MAX_VALUE - 775807, 1_000_000L),
-                Arguments.of(1, Long.MIN_VALUE + 75808, Long.MAX_VALUE - 75807, 100_000L),
-                Arguments.of(2, Long.MIN_VALUE + 5808, Long.MAX_VALUE - 5807, 10_000L),
-                Arguments.of(3, Long.MIN_VALUE + 808, Long.MAX_VALUE - 807, 1_000L),
-                Arguments.of(4, Long.MIN_VALUE + 8, Long.MAX_VALUE - 7, 100L),
-                Arguments.of(5, Long.MIN_VALUE + 8, Long.MAX_VALUE - 7, 10L),
-                Arguments.of(6, Long.MIN_VALUE, Long.MAX_VALUE, 1L));
+                Arguments.of(4, Long.MIN_VALUE, Long.MAX_VALUE, 100_000_000),
+                Arguments.of(5, Long.MIN_VALUE, Long.MAX_VALUE, 10_000_000),
+                Arguments.of(6, Long.MIN_VALUE, Long.MAX_VALUE, 1_000_000),
+                Arguments.of(7, Long.MIN_VALUE, Long.MAX_VALUE, 100_000),
+                Arguments.of(8, Long.MIN_VALUE, Long.MAX_VALUE, 10_000),
+                Arguments.of(9, Long.MIN_VALUE, Long.MAX_VALUE, 1_000),
+                Arguments.of(10, Long.MIN_VALUE, Long.MAX_VALUE, 100),
+                Arguments.of(11, Long.MIN_VALUE, Long.MAX_VALUE, 10),
+                Arguments.of(12, Long.MIN_VALUE, Long.MAX_VALUE, 1));
```

Comment:
The parameter names (minValue & maxValue) and the actual values don't match.

Please change `long` to `LongTimestampWithTimeZone`.